cmsocial/scripts/update_score.py

```diff
@@ -57,7 +57,10 @@ def compute_smart_score(user_to_task, task_to_user):
                 abilities[uid] += difficulties[tid] * score
                 num_problems += 1
             total_ability += abilities[uid]
-            user_weight[uid] = math.log(2 + 2 * num_problems) / math.log(2)
+            if num_problems < 10:
+                user_weight[uid] = 0
```

Comment:
Can we make this 1e-6 to avoid potential division by 0 issues?

Reply:
Or, I guess, replace `            if sum_score_over_ability == 0:` below with ` < 1` or something like that (that's likely better)

Reply:
Done. I guess checking whether any submission exists on the task at the beginning of the difficulty pass is now unnecessary since that case should now be handled by the check on `total_weight`.

Reply:
Indeed, that should not be necessary anymore. Nothing wrong leaving the check there though :)